### ISSUE TITLE
Function with tag result or global variables as arguments is used before definition, forcing reparse

### ISSUE THREAD

**Issue by lJohhnyl** [Zeex/pawn#146]
I updated to version 3.10.1 and there were such errors:
```
pawno\include\YSI_Coding\..\YSI_Storage\y_amx.inc(1012) : warning 208: function with tag result or global variables as arguments is used before definition, forcing reparse
pawno\include\YSI_Storage\..\YSI_Coding\y_inline/impl.inc(1468) : warning 208: function with tag result or global variables as arguments is used before definition, forcing reparse
pawno\include\YSI_Storage\..\YSI_Coding\y_inline/impl.inc(1630) : warning 208: function with tag result or global variables as arguments is used before definition, forcing reparse
```
In earlier versions, this was not.
 P.S:
```
//y_amx(1012)
stock AMX_ReadString(addr, str[], len = sizeof (str))
{
	new
		buffer,
		idx;
	do
	{
		// Read 4 bytes.
		#emit LREF.S.pri                addr
		#emit STOR.S.pri                buffer
		// Write PACKED strings.
		buffer = (buffer >>> 24) | (buffer >> 8 & 0x0000FF00) | (buffer << 8 & 0x00FF0000) | (buffer << 24);
		str[idx] = buffer;
		if (!(buffer & 0x000000FF && buffer & 0x0000FF00 && buffer & 0x00FF0000 && buffer & 0xFF000000))
		{
			return;
		}
		addr += 4;
	}
	while (++idx < len);
}
//impl.inc(1468)
static stock Remote_DoSearch(const str[], ptr)
{
	// Given a start pointer and a string, find the pointer before this string.
	new
		result;
	while (ptr != -1)
	{
		#emit PUSH.C     0x7FFFFFFF
		#emit PUSH.C     0
		#emit PUSH.S     str
		#emit LOAD.S.pri ptr
		#emit ADD.C      4
		#emit PUSH.pri
		#emit LREF.S.pri 0xFFFFFFEC
		#emit SWAP.pri
		// Compare the strings.
		#emit PUSH.C     16
		#emit SYSREQ.C   strcmp
		#emit STACK      20
		#emit STOR.S.pri result
		// Result found.  Return.
		if (!result) return ptr;
		{}
		// Didn't find it yet, try the next one.
		#emit LREF.S.pri ptr
		#emit STOR.S.pri ptr
	}
	return -1;
}
//y_impl.inc(1630)
static stock Remote_WriteSpecAndFunc(/* & */ fptr, const func[], sptr, const spec[])
{
	new
		ptr = CGen_GetCodeSpace();
	// Add to the front of the list
	AMX_Write(ptr, AMX_Read(fptr)),
	AMX_Write(fptr, ptr),
	// Store the function name.
	Remote_RawStrpack(ptr + 4, func),
	sptr = AMX_Read(sptr + 4),
	AMX_Write(ptr + 8, sptr),
	// Write the code.
	CGen_AddCodeSpace(3 * 4); // Pointers.
	new
		ctx[AsmContext];
	CGen_UseCodeSpace(ctx);
	// Convert empty strings to "NULL" ("\0" -> "\1\0").
	new
		ss = 0;
	for (new i = 0, j = strlen(spec); i != j; ++i)
	{
		if (spec[i] == 's')
		{
			if (!ss++) @emit CONST.alt ref(NULL) // Get the string pointer.
			// A string, test and convert it.
			@emit LREF.S.pri (i * 4 + 16) // Load the first character.
			@emit JNZ.rel    8            // Not '\0', skip replacement.
			@emit STOR.S.alt (i * 4 + 16) // Write the new pointer to "NULL".
		}
	}
	@emit CONST.pri  sptr
	@emit CONST.alt  AMX_Read(ptr + 4)
	@emit JUMP       YSI_g_sRemoteStub // Jump to the standard stub code.
	// Skip over the newly written code and the stored specifier.
	CGen_AddCodeSpace(6 * 4 + (ss ? (ss * 6 * 4 + 2 * 4) : 0));
	// Return a direct SCTRL pointer to the code.
	return ptr + 3 * 4 - AMX_HEADER_COD;
}
```

**Comment by Y-Less** [comment]
https://github.com/Misiur/YSI-Includes/issues/154

**Comment by Zeex** [comment]
This was fixed in 3.10.2 released today: https://github.com/Zeex/pawn/releases/tag/v3.10.2
